plugins/okta/komand_okta/tasks/monitor_logs/task.py

```diff
@@ -64,22 +65,71 @@ def run(self, params={}, state={}):  # pylint: disable=unused-argument
     def get_current_time():
         return datetime.now(timezone.utc)
 
+    @staticmethod
+    def get_iso(time: datetime) -> str:
+        """
+        Match the timestamp format used in old collector code and the format that Okta uses for 'published' to allow
+        comparison in `get_events`. e.g. '2023-10-02T15:43:51.450Z'
+        :param time: newly formatted time string value.
+        :return: formatted time string.
+        """
+        return time.isoformat("T", "milliseconds").replace("+00:00", "Z")
+
     @staticmethod
     def get_next_page_link(headers: dict) -> str:
+        """
+        Find the next page of results link from the response headers. Header example: `link: <url>; rel="next"`
+        :param headers: response headers from the request to Okta.
+        :return: next page link if available.
+        """
         links = headers.get("link").split(", ")
         next_link = None
         for link in links:
             matched_link = re.match("<(.*?)>", link) if 'rel="next"' in link else None
             next_link = matched_link.group(1) if matched_link else None
         return next_link
 
+    def get_events(self, logs: list, time: str) -> list:
+        """
+        In the collector code we would iterate over all events and drop any that match the 'since' parameter to make
+        sure that we don't double ingest the same event from the previous run (see `get_last_collection_timestamp`).
+        :param logs: response json including all returned events from Okta.
+        :param time: 'since' parameter being used to query Okta.
+        :return: filtered_logs: removed any events that matched the query start time.
+        """
+
+        # If Okta returns only 1 event (no new events occurred) returned in a previous run we want to remove this.
+        if len(logs) == 1 and logs[0].get("published") == time:
+            self.logger.info("No new events found since last execution.")
+            return []
+
+        log = "Returning {filtered} log event(s) from this iteration. "
+        pop_index, filtered_logs = 0, logs
+
+        for index, event in enumerate(logs):
+            published = event.get("published")
+            if published and published > time:
+                pop_index = index
+                break
+        if pop_index:
+            filtered_logs = logs[pop_index:]
+            log += f"Removed {pop_index} event log(s) that should have been returned in previous iteration."
```

Comment:
The blank space at the end of the first log, should be at the start of the second log as you only want to appended the space if the 2nd part of the log is appended to the first part

Reply:
updated :)